mParticle-Swrve/MPKitSwrve.m

```diff
@@ -0,0 +1,383 @@
+#import "MPKitSwrve.h"
+
+/* Import your header file here
+*/
+#if defined(__has_include) && __has_include(<SwrveSDK/SwrveSDK.h>)
+#import <SwrveSDK/SwrveSDK.h>
+#else
+#import "SwrveSDK.h"
+#endif
+
+@implementation MPKitSwrve
+/*
+    mParticle will supply a unique kit code for you. Please contact our team
+*/
++ (NSNumber *)kitCode {
+    return @216;
+}
+
++ (void)load {
+    MPKitRegister *kitRegister = [[MPKitRegister alloc] initWithName:@"Swrve" className:@"MPKitSwrve"];
+    [MParticle registerExtension:kitRegister];
+}
+
+- (MPKitExecStatus *)execStatus:(MPKitReturnCode)returnCode {
+    return [[MPKitExecStatus alloc] initWithSDKCode:self.class.kitCode returnCode:returnCode];
+}
+
+#pragma mark - MPKitInstanceProtocol methods
+
+#pragma mark Kit instance and lifecycle
+- (MPKitExecStatus *)didFinishLaunchingWithConfiguration:(NSDictionary *)configuration {
+    NSString *appId = configuration[@"app_id"];
+    NSString *apiKey = configuration[@"api_key"];
+    if (!apiKey || !appId) {
+        return [self execStatus:MPKitReturnCodeRequirementsNotMet];
+    }
+    
+    _configuration = configuration;
+    _started=NO;
+    
+    return [self execStatus:MPKitReturnCodeSuccess];
+}
+
+- (void)start {
+    _init_called=NO;
+    static dispatch_once_t swrvePredicate;
+    
+    dispatch_once(&swrvePredicate, ^{
+        /*
+         Start your SDK here. The configuration dictionary can be retrieved from self->_configuration
+         */
+        int appId = [self.configuration[@"app_id"] intValue];
+        NSString *apiKey = self.configuration[@"api_key"];
+        SwrveConfig* config = [[SwrveConfig alloc] init];
+        MPKitAPI *kitAPI = [[MPKitAPI alloc] init];
+        FilteredMParticleUser *currentUser = [kitAPI getCurrentUserWithKit:self];
+        //MPIdentityApi* currentId = [[MParticle sharedInstance] identity];
+        NSNumber *mpid = currentUser.userId;
+        config.pushResponseDelegate = [UIApplication sharedApplication];
```

Comment:
this pushResponseDelegate may not work,  probably better to allocate this to the file itself by extending the class with  `SwrvePushResponseDelegate` and adding appropriate responses to the two functions there. seems off to put it to `[UIApplication sharedApplication]`

Reply:
also, since this is in `dispatch` I wouldn't call `UIApplication` that'll give Apple something to complain about in their static code analysis